opm/material/fluidmatrixinteractions/EclEpsTwoPhaseLaw.hpp

```diff
@@ -468,17 +468,7 @@ class EclEpsTwoPhaseLaw : public EffLawT::Traits
     template <class Evaluation>
     static Evaluation unscaledToScaledPcnw_(const Params &params, const Evaluation& unscaledPcnw)
     {
-        if (params.config().enableLeverettScaling()) {
-            // HACK: we need to divide the "unscaled capillary pressure" by the
-            // conversion factor for deck to SI pressures because this quantity is
-            // actually *not* a pressure but the (dimensionless) value of the
-            // J-function. Since opm-parser currently always treads this column as a
-            // pressure, it multiplies it with the conversion factor and we have to
-            // revert this here.
-            Scalar alpha = params.scaledPoints().leverettFactor();
-            return unscaledPcnw*(alpha/params.config().deckPressureConversionFactor());
-        }
-        else if (params.config().enablePcScaling()) {
+        if (params.config().enablePcScaling()) {
```

Comment:
hm: this does not seem to be correct; the J-function is never actually used...

(instead, it is treated as a normal pressure again. I guess this will even throw with OPM/opm-parser#970.)

to fix this, you'll probably have to do something in `EclTwoPhaseMaterial`. what exactly is the non-trivial part ;)